lib/type/number.ex

```diff
@@ -11,9 +11,115 @@ defmodule Litmus.Type.Number do
   ]
 
   @type t :: %__MODULE__{
-          min: non_neg_integer,
-          max: non_neg_integer,
+          min: non_neg_integer | nil,
+          max: non_neg_integer | nil,
           integer: boolean,
           required: boolean
         }
+
+  alias Litmus.Required
+
+  @spec validate_field(t, binary, map) :: {:ok, map} | {:error, binary}
+  def validate_field(type, field, data) do
+    with {:ok, data} <- Required.validate(type, field, data),
+         {:ok, data} <- convert(type, field, data),
+         {:ok, data} <- min_validate(type, field, data),
+         {:ok, data} <- max_validate(type, field, data),
+         {:ok, data} <- integer_validate(type, field, data) do
+      {:ok, data}
+    else
+      {:error, msg} -> {:error, msg}
+    end
+  end
+
+  @spec string_to_number(binary) :: number | :error
+  defp string_to_number(str) do
+    modified_str =
+      try do
+        String.to_integer(str)
+      rescue
+        ArgumentError ->
+          try do
+            String.to_float("0" <> str)
+          rescue
+            ArgumentError ->
+              :error
+          end
+      end
+
+    modified_str
+  end
```

Comment:
I don't think we should rely on catching errors and retrying. I would look into `Integer.parse` and `Float.parse` methods, and see if there is a better way to do this.